### PR TITLE
Trezor GPG agent

### PR DESCRIPTION
**Pull request by jb55**
###### Motivation for this change

This is a continuation of #32773 /cc @fpletz @FRidh 

See the commits for why I've moved it out of python modules.

This patchset adds a systemd user service that provides a Trezor GPG agent. This allows you to use your Trezor for signing and encrypting things.

range-diff: v1 -> v2

```diff
 1:  1e63ead03d5 < -:  ----------- trezor-agent: 0.9.0 -> 0.9.2
 2:  939215154c6 < -:  ----------- trezor-agent: support qt pinentry on non-ttys
 -:  ----------- > 1:  09558b61be6 trezor-agent: 0.9.0 -> 0.12.0
 3:  971112bd12d ! 2:  c35222cbb30 trezor-agent: system service for trezor-gpg-agent
@@ -4,6 +4,8 @@
 
     Allows you to use your Trezor as a GnuPG agent
 
+    Signed-off-by: William Casarin <jb55@jb55.com>
+
 diff --git a/nixos/modules/programs/gnupg.nix b/nixos/modules/programs/gnupg.nix
 --- a/nixos/modules/programs/gnupg.nix
 +++ b/nixos/modules/programs/gnupg.nix
@@ -27,8 +29,8 @@
 +      example = "/home/alice/.gnupg/trezor";
 +      description = ''
 +        The GNUPGHOME path for your Trezor GnuPG config. See
-+        https://github.com/romanz/trezor-agent/blob/master/README-GPG.md on how
-+        to set this up.
++        https://github.com/romanz/trezor-agent/blob/master/doc/README-GPG.md on
++        how to set this up.
 +      '';
 +    };
 +
@@ -51,7 +53,7 @@
 +        GNUPGHOME=cfg.trezor-agent.configPath;
 +      };
 +
-+      serviceConfig.ExecStart = "${pkgs.python27Packages.trezor_agent}/bin/trezor-gpg-agent";
++      serviceConfig.ExecStart = "${pkgs.trezor-agent}/bin/trezor-gpg-agent";
 +    };
 +  });
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/security/trezor-agent/default.nix
Comment (srhb) [inline pkgs/tools/security/trezor-agent/default.nix]: This fails eval. It's `stdenv.lib.maintainers.*`, not `stdenv.maintainers.*`.

Comment (srhb) [reply pkgs/tools/security/trezor-agent/default.nix]: Just to clarify, this is probably because you never tested the meta attributes locally. OfBorg does this very strictly (for exactly that reason, normally we don't evaluate these attributes when we simply build a package)

So no worries, catching this in OfBorg is completely normal. :)

**Comment by mmahut** [comment]
@jb55 any update?

**Comment by jb55** [comment]
@mmahut rebased. I use this daily, works great!

**Review by mmahut — CHANGES_REQUESTED** [review]
The agent built from this pull request don't run for me:

```
# ./trezor-gpg
Traceback (most recent call last):
  File "/nix/store/24nxjvrrga6py9l564j5i2z8hnl9qcms-python3.7-trezor_agent-0.12.0/bin/.trezor-gpg-wrapped", line 11, in <module>
    sys.exit(gpg_tool())
  File "/nix/store/24nxjvrrga6py9l564j5i2z8hnl9qcms-python3.7-trezor_agent-0.12.0/bin/trezor_agent.py", line 6, in <lambda>
    gpg_tool = lambda: libagent.gpg.main(DeviceType)
  File "/nix/store/xl421r94zlhp2skqk37rlnm7s9xwygks-python3.7-libagent-0.13.1/lib/python3.7/site-packages/libagent/gpg/__init__.py", line 286, in main
    resources_map = {r.key: r for r in pkg_resources.require(agent_package)}
  File "/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages/setuptools-41.0.1-py3.7.egg/pkg_resources/__init__.py", line 900, in require
  File "/nix/store/yf4i32dx953p2dv2agfdyxdwg6ba0l61-python3.7-setuptools-41.0.1/lib/python3.7/site-packages/setuptools-41.0.1-py3.7.egg/pkg_resources/__init__.py", line 786, in resolve
pkg_resources.DistributionNotFound: The 'wheel>=0.32.3' distribution was not found and is required by libagent
```

**Comment by jb55** [comment]
ah yeah I've been using it on unstable, something might have broke in master. Will check it out soon.

**Comment by mmahut** [comment]
Any updates?

**Comment by ivan** [comment]
For the author, reviewers, and committers: this PR was scanned and appears to add a use of the deprecated `types.string`, which emits a warning as of #66346. Before merging, please change this to another type, possibly:

- `types.str` for a single string where merging does not make sense, or cannot work
- `types.lines` for multi-line configuration or scripts where merging is possible
- `types.listOf types.str` for a mergeable list of strings

**Comment by jb55** [comment]
@mmahut I can't seem to figure out how to fix the wheel issue. I think I forgot about this because I originally ran that command from repo itself.

This is sufficiently old now so I think I'll close this PR until I get around to a fix.

**Pull request closed without merging**
